### PR TITLE
Protect against null bytes in URI

### PR DESCRIPTION
**Pull request by whit537**
Can lead to reflected XSS: https://hackerone.com/reports/262852.

### CODE REVIEW

**Comment by whit537** [comment]
I'm getting unexpected results from Travis. Latest run shows 15 failures where I expect 1. Can't repro locally. Restart split green/red just prior. Wipe cache?

https://travis-ci.org/gratipay/gratipay.com/builds/268075598
https://travis-ci.org/gratipay/gratipay.com/builds/268075583

**Comment by whit537** [comment]
Now one has 15 failures and one has 16! 😖 

screen shot 2017-08-24 at 2 19 37 pm


screen shot 2017-08-24 at 2 19 33 pm

**Comment by whit537** [comment]
Back on track! b66b99c07f418e7aecc65ba8f0d94511503f6e3c is red with the redness we expect, and 8d63ed57c3a1c836c9b030903cf7c938f83c3b39 is green.

Ready for final review and merge, @EdOverflow et al.!

**Comment by EdOverflow** [comment]
Fantastic work @whit537.

**Comment by whit537** [comment]
You, too! :-)

File: www/on/npm/%package.spt [hunk 1]
```diff
@@ -56,7 +56,7 @@ if user.participant:
 {% if user.ANON %}
     <p class="instructions">
         {{ _( 'Claim the {package_link} npm package on Gratipay:'
-            , package_link=('<a href="' + package.remote_human_url + '">' + package_name + '</a>')|safe
+            , package_link=('<a href="' + package.remote_human_url + '">' + package.name + '</a>')|safe
```
Comment (Changaco) [inline www/on/npm/%package.spt]: For the record, it's possible to inject links safely into translations, e.g.:

```
, package_link='<a href="%s">%s</a>'|safe % (package.remote_human_url, package.name)
```

That way the string passed to `|safe` is a constant, and the interpolated variables are automatically escaped (`|safe` is an alias for [markupsafe](https://github.com/pallets/markupsafe)'s `Markup()`).

Comment (whit537) [reply www/on/npm/%package.spt]: Nice, thanks! PR in #4585.

**Pull request merged**